release_process.md

```diff
@@ -0,0 +1,41 @@
+# Release process
+
+We create a full release of Lance up to every 4 weeks. In between full releases,
```

Comment:
Lets update this to 2 weeks, per discussion?

Reply:
Done